modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/store/get_store_value.es.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import fetch from './../../util/fetch.es';
+import {isString} from 'metal';
+
+const TOKEN_SERIALIZE = 'serialize://';
+
+/**
+ * Sets the Store utility fetch values
+ * @param {!String} cmd
+ * @param key
+ * @param callback
+ * @review
+ */
+
+export default function getStoreValue(cmd, key, callback) {
+	cmd = 'get';
+
+	const doAsUserIdEncoded = Liferay.ThemeDisplay.getDoAsUserIdEncoded();
+
+	const url = `${Liferay.ThemeDisplay.getPathMain()}/portal/session_click?p_auth=${
+		Liferay.authToken
+	}${
+		doAsUserIdEncoded ? '&doAsUserId=' + doAsUserIdEncoded : ''
+	}&cmd=${cmd}&key=${key}`;
+
+	fetch(url, {
+		method: 'GET'
+	})
+		.then(response => {
+			return response.text();
+		})
+		.then(response => {
+			if (!isString(response)) {
+				throw new TypeError('Response must be a string');
+			}
+
+			if (response && response.indexOf(TOKEN_SERIALIZE) === 0) {
```

Comment:
1. Is there a point to doing this? Can a single returned value be a JSON object?
1. If it's a JSON object, why are we converting it to text with `response.text()` and they parsing it with `JSON.parse`? We can just use `response.json()`
1. If we end up using `JSON.parse`, we shoud let it fail. No try/catch blocks that hide error.